internal/admin/handlers.go

```diff
@@ -0,0 +1,573 @@
+
+	// Get GatewayClasses
+	if gatewayClasses, err := getGatewayClasses(ctx); err == nil {
+		response["GatewayClasses"] = gatewayClasses
+	}
+
+	// Get Gateways
+	if gateways, err := getGateways(ctx); err == nil {
+		response["Gateways"] = gateways
+	}
+
+	// Get HTTPRoutes
+	if httpRoutes, err := getHTTPRoutes(ctx); err == nil {
+		response["HTTPRoutes"] = httpRoutes
+	}
+
+	// Get GRPCRoutes
+	if grpcRoutes, err := getGRPCRoutes(ctx); err == nil {
+		response["GRPCRoutes"] = grpcRoutes
+	}
+
+	// Get TCPRoutes
+	if tcpRoutes, err := getTCPRoutes(ctx); err == nil {
+		response["TCPRoutes"] = tcpRoutes
+	}
+
+	// Get UDPRoutes
+	if udpRoutes, err := getUDPRoutes(ctx); err == nil {
+		response["UDPRoutes"] = udpRoutes
+	}
+
+	// Get TLSRoutes
+	if tlsRoutes, err := getTLSRoutes(ctx); err == nil {
+		response["TLSRoutes"] = tlsRoutes
+	}
+
+	// Get BackendTLSPolicies
+	if backendTLSPolicies, err := getBackendTLSPolicies(ctx); err == nil {
+		response["BackendTLSPolicies"] = backendTLSPolicies
+	}
+
+	// Get BackendTrafficPolicies
+	if backendTrafficPolicies, err := getBackendTrafficPolicies(ctx); err == nil {
+		response["BackendTrafficPolicies"] = backendTrafficPolicies
+	}
+
+	// Get ClientTrafficPolicies
+	if clientTrafficPolicies, err := getClientTrafficPolicies(ctx); err == nil {
+		response["ClientTrafficPolicies"] = clientTrafficPolicies
+	}
+
+	// Get SecurityPolicies
+	if securityPolicies, err := getSecurityPolicies(ctx); err == nil {
+		response["SecurityPolicies"] = securityPolicies
+	}
+
+	// Get EnvoyPatchPolicies
+	if envoyPatchPolicies, err := getEnvoyPatchPolicies(ctx); err == nil {
+		response["EnvoyPatchPolicies"] = envoyPatchPolicies
+	}
+
+	// Get EnvoyExtensionPolicies
+	if envoyExtensionPolicies, err := getEnvoyExtensionPolicies(ctx); err == nil {
+		response["EnvoyExtensionPolicies"] = envoyExtensionPolicies
+	}
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		http.Error(w, fmt.Sprintf("Failed to encode response: %v", err), http.StatusInternalServerError)
+	}
+}
+
+// handleStats returns statistics information
+func handleStats(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var memStats runtime.MemStats
+	runtime.ReadMemStats(&memStats)
+
+	response := StatsResponse{
+		"runtime": map[string]interface{}{
+			"goroutines":   runtime.NumGoroutine(),
+			"memory_alloc": memStats.Alloc,
+			"memory_total": memStats.TotalAlloc,
+			"memory_sys":   memStats.Sys,
+			"gc_runs":      memStats.NumGC,
+			"last_gc":      time.Unix(0, int64(memStats.LastGC)).Format(time.RFC3339),
+		},
+		"system": map[string]interface{}{
+			"uptime":     time.Since(startTime).String(),
+			"cpu_count":  runtime.NumCPU(),
+			"go_version": runtime.Version(),
+		},
+	}
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		http.Error(w, fmt.Sprintf("Failed to encode response: %v", err), http.StatusInternalServerError)
+	}
+}
+
+// Helper functions to get resources
```

Comment:
can we read from watchable instead ?
it will be single Load, and we already have a schema for it https://github.com/envoyproxy/gateway/blob/6a9019506ee49e222e9e1a4f6cf713fb351e4ebe/internal/gatewayapi/resource/resource.go#L35

Reply:
this will also work for non k8s / file provider case